src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -90,6 +91,27 @@ public function __construct(array $functions)
      * @throws SyntaxError
      */
     public function parse(TokenStream $stream, array $names = [])
+    {
+        $this->lint = false;
+
+        return $this->doParse($stream, $names);
+    }
+
+    /**
+     * Static analysis.
+     *
+     * The syntax of the passed expression will be checked, but not parsed.
+     * If you want to skip checking dynamic variable names, pass `null` instead of the array.
+     *
+     * @throws SyntaxError When passed expression is invalid
+     */
+    public function lint(TokenStream $stream, ?array $names = [])
+    {
+        $this->lint = true;
+        $this->doParse($stream, $names);
```

Comment:
What about returning the list of variables found in the expression? that'd allow validating them also.

Reply:
It makes sense. For example, there may be a case in which you need to check for the mandatory use of some specific variables.

Reply:
I once again reflected on this issue.
If we returned the found variables names, why we not returned the found functions names?
This feature messes code with small impact. I think the ability to pass a list of allowed variable names is enough.

Reply:
OK, works for me.
One more comment: shouldn't this return true/false instead of return void/throw?

Reply:
Only true/false is not enough. We need exception message for deteil validation error.
If we return a string in case of an error, it will not be obvious what will be returned if expression valid.
In theory, we can return an error code. But it is not clean on which part of the line this error occurred.
Now I see the issue only in translating error messages. May need improving SyntaxError class for save original message, cursor, guess, ets.